tests/metrics/estimate_gas.py

```diff
@@ -288,8 +288,8 @@ def transact(function, transaction):
     #
     # Create policy
     #
-    policy_id_1 = os.urandom(int(Policy.POLICY_ID_LENGTH))
-    policy_id_2 = os.urandom(int(Policy.POLICY_ID_LENGTH))
+    policy_id_1 = os.urandom(int(POLICY_ID_LENGTH))
```

Comment:
Seems strange that `POLICY_ID_LENGTH` is no longer defined in the `Policy` class - any reason in particular for that?

Reply:
I think, the general guideline is that if the constant is used locally in the class, then it should be a class constant. In our case, `POLICY_ID_LENGTH` is a part of the interop protocol with Solidity, so it makes sense to have it in `eth.constants`. (That invalidates my other comment - after some thought I agree that it was a correct decision to move it there). 

There are some weird parts related to that in `Policy`. We have in `Policy.__init__()`
```
self._id = construct_policy_id(self.label, bytes(self.bob.stamp))
```
but it's not really a policy ID, and it does not seem to be used anywhere except for `repr()` (which will be confusing on debug). `construct_policy_id()` does not use `POLICY_ID_LENGTH` either. 

`POLICY_ID_LENGTH` is used in `_make_enactment_payload()`, to cut the HRAC which is already the correct length (because we use HRAC as a policy ID elsewhere). This relates to my previous comment about HRACs - wouldn't it be much more logical to just have it as a policy ID?

Reply:
> This relates to my previous comment about HRACs - wouldn't it be much more logical to just have it as a policy ID?

To clarify - you mean, remove the notion of HRAC entirely in favor of Policy ID?  

I am a bit concerned that I may have been too presumptuous to change policy ID to HRAC in this PR.  Ultimately it;s changing how policies are indexed on-chain.

Reply:
> you mean, remove the notion of HRAC entirely in favor of Policy ID?

Yep. And only create it in one place, too. Currently we have in `TreasureMap.derive_hrac()`
```
self._hrac = keccak_digest(bytes(alice_stamp) + bytes(bob_verifying_key) + label)[:HRAC_LENGTH]
```
And in `Policy.__init__` + `BlockchainPolicy._make_enactment_payload()`:
```
self.hrac = keccak_digest(bytes(self.alice.stamp) + bytes(self.bob.stamp) + self.label)[:HRAC_LENGTH]
...
bytes(self.hrac)[:self.ID_LENGTH]
```
But these methods are supposed to produce the same result, which is a policy ID.

Reply:
I agree about consolidating the derivation functions themselves, I'll do that now.  I still have two outstanding concerns:

1. I'm not completely convinced that we want to solidify the union of HRAC and Policy ID and I'd like to have a focused discussion on the implications of deprecating policy ID completely.
2. From a strictly naming point of view, I much prefer the notion of HRAC over policy ID.  It's more semantically dense and accurate since knowledge of the HRAC bytes represent some degree of access/authentication.

Reply:
> From a strictly naming point of view, I much prefer the notion of HRAC over policy ID.

That makes sense. As long as we avoid having two entities which are really the same - I don't think we used HRAC for anything other than a policy IDs, right?

Reply:
👍 . It slipped my mind that policy_id in the Solidity contract is `bytes16` and that's the association - was too focused on the fact that the hrac gets truncated in Policy python class.